contracts/Core/StakeManager.sol

```diff
@@ -69,8 +69,8 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause,
         checkEpochAndState(State.Commit, epoch, parameters.epochLength())
         whenNotPaused
     {
-        require(amount >= parameters.minStake(), "staked amount is less than minimum stake required");
         uint32 stakerId = stakerIds[msg.sender];
+        require(amount + stakers[stakerId].stake >= parameters.minStake(), "staked amount is less than minimum stake required");
```

Comment:
i think it will be more gas efficient and cleaner if the require is put in `if` and `else` blocks instead.

Reply:
In that case will it not need to require. One to check if it is new staker then it should stake should be greater than minstake. And other one while increasing stake + amount should be greater than or equal to minstake. Or else we can use one require, if staker is new it should stake atleast minStake

Reply:
Please implement the suggestion.